core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ServiceProviderBuildItem.java

```diff
@@ -87,6 +91,32 @@ public static ServiceProviderBuildItem allProvidersFromClassPath(final String se
         }
     }
 
+    /**
+     * Creates a new {@link Collection} of {@code ServiceProviderBuildItem}s for the selected artifact.
+     * It includes all the providers, that are contained in all the service interface descriptor files defined in
+     * {@code "META-INF/services/"} in the selected artifact.
+     *
+     * @param dependencies the resolved dependencies of the build
+     * @param artifactCoordinates the coordinates of the artifact containing the service definitions
+     * @return a {@link Collection} of {@code ServiceProviderBuildItem}s containing all the found service providers
+     */
+    public static Collection<ServiceProviderBuildItem> allProvidersOfDependency(
+            Collection<ResolvedDependency> dependencies,
+            ArtifactCoords artifactCoordinates) {
```

Comment:
BTW, you could take a similar approach here by adding a method that takes `ArtifactCoordsPattern` and have a single call when it's called one per coords. That could be more efficient.

Reply:
Or accept a List of ArtifactCoords, I think patterns would make more sense though, since that's how they are treated anyway.

Reply:
Do you mean using a wildcard pattern, or rather a list of `ArtifactCoordsPattern` to include multiple dependencies?

Reply:
Not a list of `ArtifactCoordsPattern`s. It should be either a single `ArtifactCoordsPattern` or a `Collection<ArtifactCoords>` from which a pattern instance will be created.

Reply:
If you go with the `Collection<ArtifactCoords>` then I suppose you could keep the current method, in case there are callers that pass in a single coords.

Reply:
Sorry, if I don't follow you completely, but how would I build an `ArtifactCoordsPattern` for multiple artifacts?
Using a wildcard pattern for the group- and/or artifact-id could lead to including unwanted resources. Am I misunderstanding anything here?

Reply:
I thought of ArtifactCoordsPattern. toPatterns(Collection<ArtifactCoords> coords)

Reply:
I see. I misunderstood your earlier comment
> Not a list of `ArtifactCoordsPattern`s
That's why I disregarded this method.
Thanks for the support. I'll have a go at it.